sdk/storage/Azure.Storage.DataMovement.Blobs/README.md

```diff
@@ -45,189 +45,195 @@ The Azure.Storage.DataMovement.Blobs library uses clients from the Azure.Storage
+    SourceConditions = new()
     {
-        CreateMode = StorageResourceCreateMode.Overwrite,
+        LeaseId = leaseId
     }
-};
+}
 
-DataTransfer transfer = await container.StartUploadDirectoryAsync(localPath, options);
-
-await transfer.AwaitCompletion();
+StorageResource virtualDirectoryResource = new BlockBlobStorageResource(
+    blockBlobClient,
+    options);
 ```
 
-Download the entire container to a local directory
-```C# Snippet:ExtensionMethodSimpleDownloadContainer
-DataTransfer transfer = await container.StartDownloadToDirectoryAsync(localDirectoryPath);
-
-await transfer.AwaitCompletion();
-```
+When resuming a transfer, a credential to Azure Storage is likely needed. Credentials are not persisted by the transfer manager. When using `BlobStorageResources.TryGetResourceProviders()` to recreate a `StorageResource` for resume, the returned provider can create the resource with a credential specified by the calling code. This allows for workflows like scoping generation of a Shared Access Signature to the given resource path. Your application should provide its own mechanism for getting the appropriate credential, represented by `GenerateMySasCredential()` in the sample below.
 
-Download a directory in the container by specifying a directory prefix
-```C# Snippet:ExtensionMethodSimpleDownloadContainerDirectory
-DataTransfer tranfer = await container.StartDownloadToDirectoryAsync(localDirectoryPath2, blobDirectoryPrefix);
+```csharp
+AzureSasCredential GenerateMySasCredential(string blobUri);
 
-await tranfer.AwaitCompletion();
+if (BlobStorageResources.TryGetResourceProviders(
+    info,
+    out BlobStorageResourceProvider blobSrcProvider,
+    out BlobStorageResourceProvider blobDstProvider))
+{
+    sourceResource ??= await blobSrcProvider?.MakeResourceAsync(
+        GenerateMySasCredential(info.SourcePath));
+    destinationResource ??= await blobSrcProvider?.MakeResourceAsync(
+        GenerateMySasCredential(info.DestinationPath));
+}
 ```
 
-Download from the container specifying more advanced options
-```C# Snippet:ExtensionMethodSimpleDownloadContainerDirectoryWithOptions
-BlobContainerClientTransferOptions options = new BlobContainerClientTransferOptions
-{
-    BlobContainerOptions = new BlobStorageResourceContainerOptions
-    {
-        DirectoryPrefix = blobDirectoryPrefix
-    },
-    TransferOptions = new TransferOptions()
-    {
-        CreateMode = StorageResourceCreateMode.Overwrite,
-    }
-};
+### Upload
 
-DataTransfer tranfer = await container.StartDownloadToDirectoryAsync(localDirectoryPath2, options);
+An upload takes place between a local file `StorageResource` as source and blob `StorageResource` as destination.
 
-await tranfer.AwaitCompletion();
-```
+Upload a block blob.
 
-### Examples using BlobContainerClient extension methods to upload and download directories.
+```csharp
+string sourceLocalPath;
+BlockBlobClient destinationBlob;
 
-Create Instance of TransferManager with Options
```

Comment:
Did I not move that into the base datamovement Readme? My mistake.

Reply:
hmm it looks like there are a few samples using transfer options. Just on the base package instead of blobs, since they are agnostic of the resource type.